GitUI/CommandsDialogs/Menus/WorkingDirectoryToolStripSplitButton.cs

```diff
@@ -0,0 +1,260 @@
+using GitCommands;
+using GitCommands.UserRepositoryHistory;
+using GitUI.CommandsDialogs.BrowseDialog;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs.Menus
+{
+    /// <summary>
+    ///  Represents a split button that contains the recent repositories.
+    /// </summary>
+    internal class WorkingDirectoryToolStripSplitButton : ToolStripSplitButton, ITranslate
+    {
+        private readonly TranslationString _noWorkingFolderText = new("No working directory");
+        private readonly TranslationString _configureWorkingDirMenu = new("&Configure this menu");
+        private readonly TranslationString _repositorySearchPlaceholder = new("Search repositories...");
+
+        private static readonly object _excludeFromFilterMarker = new();
+
+        private Func<GitUICommands>? _getUICommands;
+        private RepositoryHistoryUIService _dont_use_me_repositoryHistoryUIService;
+
+        private ToolStripMenuItem _tsmiCategorisedRepos = null!;
+        private ToolStripMenuItem _tsmiOpenLocalRepository = null!;
+        private ToolStripMenuItem _tsmiCloseRepo = null!;
+        private ToolStripMenuItem _tsmiRecentReposSettings = null!;
```

Comment:
It's part of NRT annotations, which I didn't complete (it was past midnight).
This tricks the analysers that the field is not nullable. Otherwise the field will have to be checked for null before use.

Reply:
It is a hack, which can be avoided by creating an implementation class in `Initialize` with real non-nulls and no necessity for `_use_property_instead_...`.

Reply:
Yes and no. It is an established way to tell the compiler that a given field is never a null during the life of a component. Alternatively, we'll have to create helper properties (like for `UICommands` and `RepositoryHistoryUIService`) and that's an overhead.

Reply:
I mean: Encapsulate the runtime values in a nested class `Implementation` as in dd2b46ce9034a40e336f9f97dc9c81253317a789.
I have not sorted the members yet. Some functionality could be moved to / from the `Implementation` class, too.
I have kept the sequence in the file in order to ease diffing.